src/proc/namer.rs

```diff
@@ -52,29 +55,42 @@ impl Namer {
     /// This function should be used **after** [`Namer`](crate::proc::Namer) initialization by [`reset`](Self::reset()) function.
     pub fn call_unique(&mut self, string: &str) -> String {
         let base = self.sanitize(string);
-        match self.unique.entry(base) {
+        match self.unique.entry((base, self.namespace_index)) {
             Entry::Occupied(mut e) => {
                 *e.get_mut() += 1;
-                format!("{}{}", e.key(), e.get())
+                format!("{}{}", e.key().0, e.get())
             }
             Entry::Vacant(e) => {
-                let name = e.key().to_string();
-                name
+                let name = &e.key().0;
+                if self.keywords.contains(&e.key().0) {
+                    let name = format!("{}1", name);
+                    e.insert(1);
+                    name
+                } else {
+                    name.to_string()
+                }
             }
         }
     }
 
     pub fn call(&mut self, label_raw: &str) -> String {
         let base = self.sanitize(label_raw);
-        match self.unique.entry(base) {
+        match self.unique.entry((base, self.namespace_index)) {
             Entry::Occupied(mut e) => {
                 *e.get_mut() += 1;
-                format!("{}{}", e.key(), e.get())
+                format!("{}{}", e.key().0, e.get())
             }
             Entry::Vacant(e) => {
-                let name = e.key().to_string();
-                e.insert(0);
-                name
+                let name = &e.key().0;
+                if self.keywords.contains(&e.key().0) {
```

Comment:
Hmm, so with this, we are going to scan through the keyword list on each and every token added? There can be a lot of keywords. So this may turn out to be somewhat inefficient.

Reply:
Yes, once per unique keyword. The list is not checked in `Occupied` match. I don't think how to avoid that double-check. If your unique map is no longer global, you have to check it somehow.

Reply:
Actually important destinction: This is not a list, but a hash set. That should make the lookup significantly more tolerable 😉